test/minecraft_tests.cpp

```diff
@@ -169,4 +169,61 @@ TEST_CASE("Test blocks struct") {
     CHECK((mc.getBlock(testLoc) == Blocks::AIR));
     mc.setBlock(testLoc, Blocks::STONE);
     CHECK((mc.getBlock(testLoc) == Blocks::STONE));
+}
+
+
+TEST_CASE("Test array unflattening utils") {
+    Coordinate loc1(0,0,0);
+    Coordinate loc2(4, 4, 4);
+
+    //block util tests
+    mc.setBlocks(loc1, loc2, Blocks::DIRT);
+    SUBCASE("getBlocks") {
+        auto expected = std::vector<std::vector<std::vector<BlockType>>>(
+            5,
+            std::vector<std::vector<BlockType>>(
+                5,
+                std::vector<BlockType>(
+                    5,
+                    Blocks::DIRT
+                )
+            )
+        );
+        std::vector<BlockType> blockArr = mc.getBlocks(loc1, loc2);
+        auto unFlattenedBlockArr = unFlattenBlocksArray(loc1, loc2, blockArr);
```

Comment:
This is not an optimal design for the user. If they have an option to unflatten and it would be done in essentially all cases, it should be done for them within `getBlocks`.  Same with the heights variant.